lib/metamodelutil.js

```diff
@@ -147,6 +156,13 @@ function resolveTypeNames(metaModel, table) {
         });
     }
         break;
+    case `${MetaModelNamespace}.MapDeclaration`: {
+        const nameK = metaModel.key.$class === `${MetaModelNamespace}.ObjectMapKeyType` ? metaModel.key.type.name : ModelUtil.getShortName(metaModel.key.$class);
+        metaModel.key.$class === `${MetaModelNamespace}.ObjectMapKeyType` ? metaModel.key.type.namespace = resolveName(nameK, table) : metaModel.key.namespace = resolveName(nameK, table);
+        const nameV = metaModel.value.$class === `${MetaModelNamespace}.ObjectMapValueType` ? metaModel.value.type.name : ModelUtil.getShortName(metaModel.value.$class);
+        metaModel.key.$class === `${MetaModelNamespace}.ObjectMapKeyType` ? metaModel.value.type.namespace = resolveName(nameV, table) : metaModel.value.namespace = resolveName(nameV, table);
```

Comment:
Nit, please refactor this to have a branch for ObjectMapKeyType and an else case. This will make this easier to read and DRY the repeated tenary operators.

Reply:
Looks like we may also have a bug here with `ScalarDeclaration` missing? Can you log that please?